### PR TITLE
Adds new Level: Jarvis Gone Wrong

### PR DESCRIPTION
**Pull request by dduzgun-security**
### Why:

Closes https://github.com/skills/secure-code-game/issues/16

### What's being changed:

Adding a level for GitHub Actions awareness

### Check off the following:

- [x] For workflow changes, I have verified the Actions workflows function as expected.
- [x] For content changes, I have reviewed the [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by jkcso** [comment]
@JarLob, who is an Action security specialist with research and publications of public impact, has noted the following about this level: 

_"it is truly third party untrusted action. It is good they have read permission, it is minor but if the action repo is compromised one day it still can use cpu resources for example for cryptomining. Although it means only when the game participant runs their copy of the workflow. It would be better to have the action somewhere in the Skills org, but mark it somehow as untrusted. It would also avoid breaking the game if the action is deleted one day."_

I will take this into account and work with the skills org to see if we can accommodate. However, first we need to decide formally about accepting external PRs for the game's development.

**Comment by dduzgun-security** [comment]
I agree :+1: It would indeed be a lot better if the repository containing [the action itself](https://github.com/dduzgun-security/secure-code-game-action) is in the `skills` GitHub Organization and marked as `untrusted`. If not possible, maybe a new org owned by GitHub for untrusted actions / packages instead might be an option.

**Comment by markpatterson27** [comment]
Could you not just host the action in this repo? e.g. Something like

```yaml
      - name: Check GitHub Status
        uses: ./Level-X/secure-code-game-action
        with:
          who-to-greet: "Jarvis, obviously ..."
          give-me-your-token-now: "coucou-123456789"
```

It would make maintaining the action alongside the level a lot easier.

**Comment by jkcso** [comment]
Back at it, I will have some time to work on our strategy for future levels and will merge in the upcoming few weeks!

**Comment by dduzgun-security** [comment]
@markpatterson27 thanks for the suggestion. 🙏 

> Could you not just host the action in this repo?

The idea of having it as an external action is to simulate the classic scenario (where people use GitHub Actions from the marketplace) as much as possible. This would provide extra awareness on the importance to read the GitHub Action code before using it in their pipeline. Also, the `jarvis-test.yml` looks if `uses: dduzgun-security/secure-code-game-action@v1.0` is present or not in the `jarvis.yml` to fail or pass which makes the tests a bit easier.

**Comment by dduzgun-security** [comment]
@jkcso I updated the level with clearer hints and details like we talked. Also, improved the pipeline validation tests (pass or fail) which is simpler now. Keep me posted if you see more improvements. 💪

**Review by jkcso — APPROVED** [review]

**Comment by hectorsector** [comment]
Thanks for the review @jkcso. Waiting on a review from @skills/content-reviewers and then we can get this merged in.

**Comment by jkcso** [comment]
👋🏼  Hi @JarLob - can I please have a final review here following Deniz's changes?

File: .github/workflows/jarvis-test.yml
Comment (JarLob) [inline .github/workflows/jarvis-test.yml:19]:
```suggestion
          if grep -q "uses: dduzgun-security/secure-code-game-action@" $GITHUB_WORKSPACE/.github/workflows/jarvis.yml; then
```

File: .github/workflows/jarvis.yml
Comment (JarLob) [inline .github/workflows/jarvis.yml:28]:
```suggestion
        uses: dduzgun-security/secure-code-game-action@675f79225c41e6b281d00379a52ec2364be841fe # v1.0
```

**Review by JarLob — CHANGES_REQUESTED** [review]

**Review by JarLob — APPROVED** [review]

File: README.md
Comment (cmwilson21) [inline README.md:88]:
```suggestion
Jarvis, your trusty automation butler, has been acting up lately. He's been experimenting with unofficial pipelines he found online to [get the GitHub status page](https://www.githubstatus.com/api/v2/status.json). While these actions promised to make your CI/CD pipeline faster and more efficient, they have also introduced serious security vulnerabilities into your code. Do you have what it takes to fix the bug and progress to Level 1?
```

File: README.md
Comment (cmwilson21) [inline README.md:82]: Highlighting the level - where does this fit in the game flow?

File: README.md
Comment (cmwilson21) [inline README.md:91]:
```suggestion
For each level, you will find the same file structure:
```
Updating to stay consistent with the other levels

File: README.md
Comment (cmwilson21) [inline README.md:99]:
```suggestion
1. Try to fix the bug. Do not use or install any additional dependencies to fix it.
```

**Review by cmwilson21 — APPROVED** [review]
Approving this from a content side. Thanks for quickly accepting those suggestions. Great work @dduzgun-security! This seems like a fun level! ✨ 

Currently, the new level is in the welcome detail (`<details id=0>`). I'm not sure where it best fits in with the rest of the levels, but once it's outside the welcome block, it's good to go from a content perspective! 👍

File: README.md
Comment (dduzgun-security) [reply README.md:82]: 👋  @jkcso, I will let you arrange the level number and `README.md` for the new season

**Comment by dduzgun-security** [comment]
👋  @cmwilson21 , huge thanks on the review and change recommendations. 
Really good catches! The content is now way better with the proposed fixes 🙌

**Comment by jkcso** [comment]
@cmwilson21 I'm handling this on the background through an internal repo that I will ask you to review by the mid/end September for last touches before going live, as a 2nd season for the secure code game. I will take the files from here and add them to the internal one for internal audit, then open a big PR in this public repo for a public release.

**Comment by dduzgun-security** [comment]
Added the changes to make the GitHub Action use the latest version and to make input parameters less obvious as proposed by @jkcso.

**Review by jkcso — APPROVED** [review]

**Review by jkcso — APPROVED** [review]

**Comment by jkcso** [comment]
As the second season levels are hosted in an internal repo for staff testing and content review, I am now closing this public PR. Thanks again for your valuable contribution and it will be soon merge into the public repo :-)

**Pull request closed without merging**
